build.py

```diff
@@ -214,18 +214,22 @@ def use_python_env(folder):
     step_end()
 
 if build_cli:
-    step_start("Building the command line compiler")
-    cargo_build_args = ["cargo", "build"]
-    if build_type == "release":
-        cargo_build_args.append("--release")
-    subprocess.run(cargo_build_args, check=True, text=True, cwd=root_dir)
-
     if run_tests:
-        print("Running tests for the command line compiler")
+        step_start("Running Rust unit tests")
         cargo_test_args = ["cargo", "test"]
         if build_type == "release":
             cargo_test_args.append("--release")
+            # Disable LTO for release tests to speed up compilation
+            cargo_test_args.append("--config")
+            cargo_test_args.append('profile.release.lto="off"')
         subprocess.run(cargo_test_args, check=True, text=True, cwd=root_dir)
+        step_end()
+
+    step_start("Building the command line compiler")
+    cargo_build_args = ["cargo", "build", "--bin", "qsc"]
```

Comment:
Do we need `qci`?

Reply:
No, nothing actually uses it in CI or production. At this point it's still around just because some folks like using it locally during dev/test, so they won't let me delete it ;)

Reply:
I don't love this middle ground - if the build script doesn't build it, it's effectively dead code.... the people who _don't_ use it are bound to break it for the next person who _does_ use it.

If it's a maintenance headache at this point, I'd be ok deleting it, but that's easy for me to say right now 😄